### PR TITLE
[Tech] Terminology for Decoupling

### PR DESCRIPTION
**Pull request by petorre**
Terminology For Decoupling Applications From Infrastructure, Platform and Other Application Functionality. It was before in #789 that I will close.

### CODE REVIEW

File: doc/tech/glossary.md [hunk 1]
```diff
@@ -131,9 +131,20 @@ Terms not defined by Kubernetes:
 
 **Software Defined Storage (SDS)** architecture consists of the storage software that is independent from the underlying storage hardware. The storage access software provides data request interfaces (APIs) and the SDS controller software provides storage access services and networking.
 
-
 <a name="1.6"></a>
-## 1.6 Other Referenced Terminology
+## 1.6 Terminology For Decoupling Applications From Infrastructure, Platform and Other Application Functionality:
+- Decoupling, Loose Coupling = Loosely coupled system is one in which each of its components has, or makes use of, little or no knowledge of the definitions of other separate components. Loose coupling is the opposite of tight coupling. ([Wikipedia:Loose Coupling](https://en.wikipedia.org/wiki/Loose_coupling))
```
Comment (pgoyal01) [inline doc/tech/glossary.md]: @petorre I realise this is from wikipedia but can we qualify "definitions" as APIs could qualify as "definitions"?

**Comment by ASawwaf** [comment]
@peterwoerndle , it will be great, if you can define the decoupling and its difference between abstraction as per last NFG call

File: doc/tech/glossary.md [hunk 2]
```diff
@@ -131,9 +131,20 @@ Terms not defined by Kubernetes:
...
+- Abstraction = Process of removing smaller details or attributes in the study of systems to focus attention on topics of greater importance. It can be result of decoupling. ([Wikipedia:Abstraction](https://en.wikipedia.org/wiki/Abstraction_(computer_science)))
```
Comment (pgoyal01) [inline doc/tech/glossary.md]: Would it help to also include a sentence from generalization to maybe reword as:
Process of removing smaller details or attributes or common properties in the study of systems to focus attention on topics of greater importance or general concepts.

File: doc/tech/glossary.md [hunk 3]
```diff
@@ -131,9 +131,20 @@ Terms not defined by Kubernetes:
...
+- Encapsulation = Restricting of direct access to some of an object's components. ([Wikipedia:Encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)))
+- Appliance deployment model = Application has tight coupling with underlying Platform even if the application is virtualized or containerized.
+- Tenant/Cloud deployment model = Tenant brings applications that are decoupled from the platform provided by Cloud operator. Tenant and Cloud operator are normally different legal entities.
```
Comment (pgoyal01) [inline doc/tech/glossary.md]: I don't think we need "Tenant" 
Cloud deployment model = Applications are decoupled from the platform provided by the Cloud operator.

File: doc/tech/glossary.md [hunk 4]
```diff
@@ -131,9 +131,20 @@ Terms not defined by Kubernetes:
...
+- Application Control = Any way of controlling tenant’s application. Depending on RA and technologies used, it can be tenant’s VNFM or Orchestration SW, or Platform capability.
```
Comment (pgoyal01) [inline doc/tech/glossary.md]: Again, can make it Tenant independent as in case of VNF's the VNFM/NFVO can be provided as Platform capability, by VNF vendor or by Tenant.   
Application Control = Any method or system of controlling applications (VNFs). Depending on RA and technologies used, this can be a VNF Manager or NFV Orchestrator provided as a VNF or Platform capability.

File: doc/tech/glossary.md (hunk 1 above)
Comment (petorre) [reply doc/tech/glossary.md]: Changed to "implementation details".

File: doc/tech/glossary.md (hunk 2 above)
Comment (petorre) [reply doc/tech/glossary.md]: Done.

File: doc/tech/glossary.md (hunk 3 above)
Comment (petorre) [reply doc/tech/glossary.md]: Done.

File: doc/tech/glossary.md (hunk 4 above)
Comment (petorre) [reply doc/tech/glossary.md]: Done.

**Comment by rabiabdel** [comment]
@petorre slight edit suggestion made to title and ToC directly into the file. let me know if it is all OK.

**Comment by petorre** [comment]
> 
> 
> @petorre slight edit suggestion made to title and ToC directly into the file. let me know if it is all OK.

Yes. I corrected "Abstractioin" -> " Abstraction".
